### PR TITLE
Some Empty Dataset Handling

### PR DESCRIPTION
**Pull request by xizi-xu**
- Display a message when nothing is returned from TSDB or the expression does not exist.
- Show local timezone information in the time range label.

### CODE REVIEW

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 1]
```diff
@@ -914,41 +941,39 @@ angular.module('argus.directives.charts.lineChart', [])
             }
 
             //extent, k is the least number of points in one line you want to see on the main chart view
-            function setZoomExtent(k){
-              //TODO: empty data issue
-                if(!currSeries || currSeries.length === 0) return;
-                var numOfPoints= currSeries[0].data.length;
+            function setZoomExtent(k) {
+                var numOfPoints = currSeries[0].data.length;
```
Comment (prestonfff) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: what if currSeries is []?

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 2]
```diff
@@ -974,21 +999,29 @@ angular.module('argus.directives.charts.lineChart', [])
...
-
-            // initialize starting point for graph settings & info
-            addOverlay();
-            updateDateRange();
-            enableBrushTime();
-            reset();    //to remove the brush cover first for user the drag
+            // create graph only when there is data
+            if (!series || series.length === 0) {
+                console.log("Empty data from chart data processing");
+                return;
```
Comment (prestonfff) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: So There will not be a empty chart place holder? Maybe a placeholder chart with "Empty data from chart data processing" info in the center?

**Comment by prestonfff** [comment]
Maybe we can do a separate commit for all the reformatting of codes, just to make it faster to locate the real changes and make reviewing easier.

File: ArgusWeb/app/js/directives/charts/lineChart.js (hunk 1 above)
Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: This function is only used in updateGraph. currSeries and series are checked for empty before updateGraph is called. No need to check for it again

File: ArgusWeb/app/js/directives/charts/lineChart.js (hunk 2 above)
Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: This technically should never happen. It's just a place holder for now.

**Pull request merged**